### PR TITLE
Improve REST API SQL query logging

### LINKED ISSUE #7435: Improve REST api SQL query logging
### Problem

The convention in the javascript rest api module about SQL query and params logging is a log statement at trace level is put before the SQL is executed. It provides info for debugging often times.

However it's cumbersome to remember and implement the convention,  so it's common for some queries such trace level logging is just missing.

### Solution

To make it maintenance free, we can move the trace level logging to utils.js `Pool.prototype.queryQuietly`, we can also create an `Error` object to to find the function which calls `queryQuietly`   

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by jascks**
**Description**:
The convention in the javascript rest api module about SQL query and params logging is a log statement at trace level is put before the SQL is executed. It provides info for debugging often times.

However it's cumbersome to remember and implement the convention,  so it's common for some queries such trace level logging is just missing.

To make it maintenance free, we can move the trace level logging to `utils.js` `Pool.prototype.queryQuietly`, we can also create an `Error` object to to find the function which calls `queryQuietly`  

This PR:
1. Removes all explicit logging of the query and parameter values prior to invoking `pool.queryQuietly()`.
2. In `util.js Pool.prototype.queryQuietly`, if trace logging is enabled, then an `Error` object is used to obtain the stack trace and identify the caller file name and line number which is emitted along with the SQL query and parameter values.
3. Potentially more depending to the answers to my questions on #7435.

**Related issue(s)**: 

Fixes #7435

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/utils.js [hunk 1]
```diff
@@ -1437,6 +1437,19 @@ const getPoolClass = async (mock = false) => {
...
+    if (logger.isTraceEnabled()) {
+      const callerInfo = new Error().stack
+        .split('at ', 3)
+        .map((entry) => {
+          const result = entry.match(/\/hedera-mirror-rest\/(.*)\.js:(\d+):.*/);
+          return result && result.length === 3 && {name: result[1], line: result[2]};
+        })
+        .filter((r) => !!r)
+        .filter((r) => r.name !== 'utils')[0];
+      logger.trace(`${callerInfo.name}:${callerInfo.line} query: ${query} ${utils.JSONStringify(params)}`);
```
Comment (xin-hedera) [inline hedera-mirror-rest/utils.js:1450]: there is no `utils.JSONStringify` in the `utils` module itself

Comment (jascks) [reply hedera-mirror-rest/utils.js:1450]: Funny how this did not stand out when I was running spec tests, but when running the API application in IDEA, there is a very obvious `ReferenceError`. Next push will have this fixed.

File: hedera-mirror-rest/utils.js [hunk 2]
```diff
@@ -1437,6 +1437,19 @@ const getPoolClass = async (mock = false) => {
     const clientErrorCallback = (error) => {
       logger.error(`error event emitted on pg pool. ${error.stack}`);
     };
+
+    if (logger.isTraceEnabled()) {
+      const callerInfo = new Error().stack
+        .split('at ', 3)
+        .map((entry) => {
+          const result = entry.match(/\/hedera-mirror-rest\/(.*)\.js:(\d+):.*/);
```
[the remark this answers is not in the document]
Comment (jascks) [reply hedera-mirror-rest/utils.js]: I have updated this bit of code. It continues to retrieve the file name and line number and now also the function name. So, for example, with this simple stack trace:
```
Error
    at Pool.queryQuietly (file:///Users/user/Development/mirrornode/repos/hedera-mirror-node/hedera-mirror-rest/utils.js:1442:24)
    at getOneAccount (file:///Users/user/Development/mirrornode/repos/hedera-mirror-node/hedera-mirror-rest/accounts.js:479:30)
    at async wrappedMiddleware (/Users/user/Development/mirrornode/repos/hedera-mirror-node/hedera-mirror-rest/node_modules/@awaitjs/express/index.js:64:9)
```
Output is:
```
2024-01-16T13:34:37.362-07:00 TRACE 5f4a1584 getOneAccount (accounts.js:479) query: ...
```

For something utilizing `baseServices`.
```
Error
    at Pool.queryQuietly (file:///Users/user/Development/mirrornode/repos/hedera-mirror-node/hedera-mirror-rest/utils.js:1442:24)
    at NftService.getRows (file:///Users/user/Development/mirrornode/repos/hedera-mirror-node/hedera-mirror-rest/service/baseService.js:71:31)
    at NftService.getNfts (file:///Users/user/Development/mirrornode/repos/hedera-mirror-node/hedera-mirror-rest/service/nftService.js:150:30)
    at getNftsByAccountId (file:///Users/user/Development/mirrornode/repos/hedera-mirror-node/hedera-mirror-rest/controllers/accountController.js:107:48)
    at async wrappedMiddleware (/Users/user/Development/mirrornode/repos/hedera-mirror-node/hedera-mirror-rest/node_modules/@awaitjs/express/index.js:64:9)
```
The output is:
```
2024-01-16T13:18:29.761-07:00 TRACE 0e04e895 NftService.getNfts (nftService.js:150) query: select ...
```
This is the first entry not in either `utils.js` or `baseService.js`. And the same for this stack trace:
```
Error
    at Pool.queryQuietly (file:///Users/user/Development/mirrornode/repos/hedera-mirror-node/hedera-mirror-rest/utils.js:1442:24)
    at EntityService.getRows (file:///Users/user/Development/mirrornode/repos/hedera-mirror-node/hedera-mirror-rest/service/baseService.js:71:31)
    at EntityService.getSingleRow (file:///Users/user/Development/mirrornode/repos/hedera-mirror-node/hedera-mirror-rest/service/baseService.js:79:29)
    at EntityService.isValidAccount (file:///Users/user/Development/mirrornode/repos/hedera-mirror-node/hedera-mirror-rest/service/entityService.js:90:32)
    at getTokenRelationships (file:///Users/user/Development/mirrornode/repos/hedera-mirror-node/hedera-mirror-rest/controllers/tokenController.js:82:48)
    at async wrappedMiddleware (/Users/user/Development/mirrornode/repos/hedera-mirror-node/hedera-mirror-rest/node_modules/@awaitjs/express/index.js:64:9)
```
With output:
```
2024-01-16T13:31:29.343-07:00 TRACE 14b7a006 EntityService.isValidAccount (entityService.js:90) query: select type
```
Note that both `EntityService` and `NftService` are classes and that must explain the naming in the stack trace.

File: hedera-mirror-rest/utils.js (hunk 1 above)
Comment (jascks) [reply hedera-mirror-rest/utils.js]: Taken care of.

File: hedera-mirror-rest/service/baseService.js [hunk 3]
```diff
@@ -65,10 +65,9 @@ class BaseService {
     };
   }
 
+  // TODO depending on feedback on issue 7435 perhaps functionName may be provided to queryQuietly() or go away completely.
+  // TODO perhaps still useful for row count logging done here explicitly?
```
Comment (jascks) [inline hedera-mirror-rest/service/baseService.js:69]: I'm curious about your thoughts on this. My inclination is to remove the `functionName` parameter and modify all calling code to remove it their as well.

Is the remaining logging statement to report the number of entries returned still useful (w/o `functionName`)? If not, then `getRows()` itself is not very useful .

Comment (xin-hedera) [reply hedera-mirror-rest/service/baseService.js:69]: agree with the logic in queryQuietly, we no longer need the `functionName` parameter in either `getRows` or `getSingleRow`

File: hedera-mirror-rest/utils.js [hunk 4]
```diff
@@ -1437,6 +1437,21 @@ const getPoolClass = async (mock = false) => {
     const clientErrorCallback = (error) => {
       logger.error(`error event emitted on pg pool. ${error.stack}`);
     };
+
+    if (logger.isTraceEnabled()) {
+      const callerInfo = new Error().stack
+        .split('at ')
```
Comment (xin-hedera) [inline hedera-mirror-rest/utils.js:1443]: better change to the code snippet I proposed. We'll get surprises if we ever name a function ending with `at `...

File: hedera-mirror-rest/service/baseService.js (hunk 3 above)
Comment (jascks) [reply hedera-mirror-rest/service/baseService.js]: Cool, thank you. This has been done.

File: hedera-mirror-rest/utils.js [hunk 5]
```diff
@@ -1437,6 +1437,21 @@ const getPoolClass = async (mock = false) => {
     const clientErrorCallback = (error) => {
       logger.error(`error event emitted on pg pool. ${error.stack}`);
     };
+
+    if (logger.isTraceEnabled()) {
+      const callerInfo = new Error().stack
+        .split('\n')
+        .splice(1)
+        .filter((s) => !(s.includes('utils.js') || s.includes('baseService.js')))
```
Comment (jascks) [inline hedera-mirror-rest/utils.js:1445]: It looks like `.splice(1)` discards the first element which is `Error`. However, `.trim()` is an unresolved function. It does not seem to be a thing for `Array`. But, the regex in `map()` ignores the leading whitespace.

File: hedera-mirror-rest/utils.js [hunk 6]
```diff
@@ -1437,6 +1437,21 @@ const getPoolClass = async (mock = false) => {
...
+        .map((entry) => {
+          const result = entry.match(/^\s*at\s+(\S+).*\/(.*\.js):(\d+):.*/);
+          return result && result.length === 4 && {function: result[1], file: result[2], line: result[3]};
+        })[0];
```
Comment (jascks) [inline hedera-mirror-rest/utils.js:1449]: With `splice()` and `filter()` discarding the unwanted stuff from the top of the stack, `[0]` and not `[1]` provides the information I need to log.

**Review by xin-hedera — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Pull request merged**